### PR TITLE
Feature cont.: authorize CLI as admin with private 

### PR DESCRIPTION
**Pull request by noiarek**
**What changed?**
`--jwt-private-key` flag added to CLI

**Why?**
Requested in [this proposal](https://docs.google.com/document/d/1a-Xo9FSCLubYlIQHB_eX5j4Yv2GDv_82J94At2dAL3o/edit#heading=h.iar4if2w5gus)

**How did you test it?**
Tested locally
```
./cadence-server --env development_oauth start
```
and
```
./cadence --do samples-domain --jwt-private-key config/credentials/keytest domain desc
```

**Potential risks**
Nothing

**Release notes**
When the whole proposal is implemented

**Documentation Changes**
When the whole proposal is implemented

### CODE REVIEW

File: common/rsa.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+type KeyType string
+
+const (
+	KeyTypePrivate KeyType = "private key"
+
+	KeyTypePublic KeyType = "public key"
+)
+
+func loadRSAKey(path string, keyType KeyType) (interface{}, error) {
```
Comment (longquanzheng) [inline common/rsa.go:40]: 👍👍👍

**Review by longquanzheng — APPROVED** [review]
Looks great

File: tools/cli/app.go [hunk 2]
```diff
@@ -66,9 +66,14 @@ func NewCliApp() *cli.App {
 		},
 		cli.StringFlag{
 			Name:   FlagJWT,
-			Usage:  "optional JWT for authorization",
+			Usage:  "optional JWT for authorization. Either this or --jwt-private-key is needed for jwt authentication",
 			EnvVar: "CADENCE_CLI_JWT",
 		},
+		cli.StringFlag{
+			Name:   FlagJWTPrivateKeyWithAlias,
+			Usage:  "optional private key path to create JWT. Either this or --jwt is needed for jwt authentication. --jwt flag has priority over this one if both provided",
```
Comment (noiarek) [inline tools/cli/app.go:74]: I've never been quite good with text description @longquanzheng is this decent enough? :laughing:

Comment (longquanzheng) [reply tools/cli/app.go:74]: Looks great. Thanks 😊

Comment (longquanzheng) [reply tools/cli/app.go:74]: Ohh can you say authorization instead of authentication?

Comment (longquanzheng) [reply tools/cli/app.go:74]: This feature doesn’t verify the user identity. The provided jwt token is to verify the access. I think by concept the authentication part (using username password or one login) is done in the OAuth integration by the user.

Comment (noiarek) [reply tools/cli/app.go]: This is my classic typo. Updated :)

**Pull request merged**